paasta_tools/monitoring/check_mesos_outdated_tasks.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+# Copyright 2015-2017 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+import sys
+import time
+
+from paasta_tools.mesos_tools import get_mesos_master
+from paasta_tools.utils import load_system_paasta_config
+
+
+OUTPUT_FORMAT = "{:<30}  {:<8}  {:<20}  {:<27}  {}"
+
+
+def get_mesos_state():
+    state = get_mesos_master().state
+    return state
+
+
+def marathon_tasks(state):
+    for framework in state.get('frameworks', []):
+        if framework['name'].lower().startswith('marathon'):
+            for task in framework.get('tasks', []):
+                yield task
+
+
+def create_slave_id_to_hostname_dict(state):
+    res = {}
+    for slave in state['slaves']:
+        res[slave['id']] = slave['hostname']
+    return res
+
+
+def group_running_tasks_by_job_id_and_gitsha(state):
+    res = {}
+    for t in marathon_tasks(state):
+        if t['state'] == 'TASK_RUNNING':
+            job_id = t['name'][:t['name'].find('.', t['name'].find('.') + 1)]
```

Comment:
don't we already have functions in marathon_tools to do this?

Reply:
We have marathon_tools.parse_service_instance_from_executor_id that returns a tuple of srv_name, srv_instance. Then I still need to concatenate them and to get gitsha from the same task_id.
It looks more complex than the current implementation.